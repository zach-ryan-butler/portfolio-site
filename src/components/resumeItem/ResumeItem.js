import React from 'react';
import { Card, CardContent, Box, Typography } from '@material-ui/core';
import CalendarTodayIcon from '@material-ui/icons/CalendarToday';
import PropTypes from 'prop-types';

import useStyles from './ResumeItem.styles';

function ResumeItem(props) {
  const classes = useStyles();
  return (
    <Card className={classes.root}>
      <CardContent>
        <Box className={classes.dateContainer}>
          <CalendarTodayIcon className={classes.icon} />
          <Typography variant='subtitle2' className={classes.dateText}>
            {props.date}
          </Typography>
        </Box>
        <Typography className={classes.whereText}>{props.where}</Typography>
        <Typography variant='body2' className={classes.titleText}>
          {props.title}
        </Typography>
        <Typography variant='body2' className={classes.bodyText}>
          {props.children}
        </Typography>
      </CardContent>
    </Card>
  );
}

ResumeItem.propTypes = {
  date: PropTypes.string,
  title: PropTypes.string,
  children: PropTypes.string,
  where: PropTypes.string,
};

export default ResumeItem;
