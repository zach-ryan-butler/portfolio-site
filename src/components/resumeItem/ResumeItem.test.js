import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ResumeItem from './ResumeItem';

describe('Resume item component', () => {
  afterEach(() => cleanup());

  it('takes a snapshot of the component', () => {
    const { asFragment } = render(<ResumeItem props={{}} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
