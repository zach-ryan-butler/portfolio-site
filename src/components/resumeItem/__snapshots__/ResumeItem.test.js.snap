// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resume item component takes a snapshot of the component 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root"
    >
      <div
        class="MuiBox-root MuiBox-root-8 makeStyles-dateContainer-3"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-icon-5"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 3h-1V1h-2v2H7V1H5v2H4c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 18H4V8h16v13z"
          />
        </svg>
        <h6
          class="MuiTypography-root makeStyles-dateText-4 MuiTypography-subtitle2"
        />
      </div>
      <p
        class="MuiTypography-root makeStyles-whereText-7 MuiTypography-body1"
      />
      <p
        class="MuiTypography-root makeStyles-titleText-6 MuiTypography-body2"
      />
      <p
        class="MuiTypography-root makeStyles-bodyText-2 MuiTypography-body2"
      />
    </div>
  </div>
</DocumentFragment>
`;
